src/api/client/Assets.ts

```diff
@@ -250,4 +257,77 @@ export class Assets {
       next,
     };
   }
+
+  /**
+   * Retrieve all the Asset Global Metadata on chain.
+   * This includes metadata id, name and optional specs
+   */
+  public async getGlobalMetadata(): Promise<AssetGlobalMetadata[]> {
+    const {
+      context: {
+        polymeshApi: {
+          query: {
+            asset: { assetMetadataGlobalKeyToName, assetMetadataGlobalSpecs },
+          },
+        },
+      },
+    } = this;
+
+    const [keyToNameEntries, specsEntries] = await Promise.all([
+      assetMetadataGlobalKeyToName.entries(),
+      assetMetadataGlobalSpecs.entries(),
+    ]);
+
+    return keyToNameEntries.map(
+      ([
+        {
+          args: [rawId],
+        },
+        rawName,
+      ]) => {
+        const metadata: AssetGlobalMetadata = {
+          id: u64ToBigNumber(rawId),
+          name: bytesToString(rawName.unwrap()),
+        };
+
+        const specEntry = specsEntries.find(
```

Comment:
Asked this on Slack. If so, it will be more performant to use the index offset directly instead of iterating with find on every loop. Since we have lodash there is `zip` that could group them if they are ordered the same.

Otherwise build up a `Record<rawId, Spec>` before this loop.

With `find` in map this becomes O(n<sup>2</sup>), which I think we can avoid and make this function more readable.


https://polymathnetwork.slack.com/archives/C01RLA867C5/p1660750728813059